tests/test_sensor.py

```diff
@@ -82,47 +86,80 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
-def test_sensor_value_setter(diverse_sensors):
-    """Check whether the value type verification works."""
-    diverse_sensors["test-bool-sensor"].value = True
-    assert diverse_sensors["test-bool-sensor"]
-    with pytest.raises(TypeError):
-        diverse_sensors["test-bool-sensor"].value = "True"
-
-    diverse_sensors["test-int-sensor"].value = 1234
-    assert diverse_sensors["test-int-sensor"].value == 1234
-    with pytest.raises(TypeError):
-        diverse_sensors["test-int-sensor"].value = "1234"
+class MyEnum(enum.Enum):
+    ZERO = 0
+    ONE = 1
 
-    diverse_sensors["test-float-sensor"].value = 1234.5
-    assert diverse_sensors["test-float-sensor"].value == 1234.5
-    with pytest.raises(TypeError):
-        diverse_sensors["test-float-sensor"].value = "1234"
 
-    diverse_sensors["test-str-sensor"].value = "foo"
-    assert diverse_sensors["test-str-sensor"].value == "foo"
-    with pytest.raises(TypeError):
-        diverse_sensors["test-str-sensor"].value = 1234
+class OtherEnum(enum.Enum):
+    ABC = 0
+    DEF = 1
 
-    diverse_sensors["test-bytes-sensor"].value = b"bar"
-    assert diverse_sensors["test-bytes-sensor"].value == b"bar"
-    with pytest.raises(TypeError):
-        diverse_sensors["test-bytes-sensor"].value = "foo"
 
-    diverse_sensors["test-address-sensor"].value = "1.2.3.4"
-    assert diverse_sensors["test-address-sensor"].value == Address("1.2.3.4")
-    with pytest.raises(TypeError):
-        diverse_sensors["test-address-sensor"].value = 5678
+@pytest.mark.parametrize(
+    "sensor_name,sensor_type,default,compatible_value",
+    [
+        ("bool", bool, False, True),
+        ("int", int, 0, np.int32(1234)),
+        ("float", float, 0.0, np.int32(1234678)),
+        ("float", float, 0.0, np.float32(12345678)),
+        ("str", str, "zero", "one-two-three-four"),
+        ("bytes", bytes, b"zero", b"one-two-three-four"),
+        ("address", Address, Address(IPv4Address("0.0.0.0")), Address(IPv4Address("1.2.3.4"))),
+        ("timestamp", Timestamp, Timestamp(0.0), 12345678),
+        ("enum", MyEnum, MyEnum.ZERO, MyEnum.ONE),
+    ],
+)
+def test_sensor_value_setter_success(
+    sensor_name: str,
+    sensor_type: Type[_T],
+    default: Any,
+    compatible_value: Any,
+) -> None:
+    """Check a compatible value against a sensor type.
+
+    This uses a default value to ensure the :class:`Sensor`\'s constructor
+    also performs the value type check. It also checks compatibility with
+    numpy scalars.
+    """
+    sensor_under_test = Sensor(
```

Comment:
`sensor_under_test` is a bit verbose - you could just call it `sensor` and it wouldn't be confused with anything else.